src/components/Dialog/ImportSongProjectDialog.vue

```diff
@@ -0,0 +1,295 @@
+          placeholder="外部プロジェクトファイルを選択してください"
+          @input="handleFileChange"
+        />
+        <QSelect
+          v-if="project"
+          v-model="selectedTrack"
+          :options="tracks"
+          :disable="projectFileError != undefined"
+          emitValue
+          mapOptions
+          label="インポートするトラック"
+        />
+      </QPageContainer>
+      <QFooter>
+        <QToolbar>
+          <QSpace />
+          <QBtn
+            unelevated
+            align="right"
+            label="キャンセル"
+            color="toolbar-button"
+            textColor="toolbar-button-display"
+            class="text-no-wrap text-bold q-mr-sm"
+            @click="handleCancel"
+          />
+          <QBtn
+            unelevated
+            align="right"
+            label="インポート"
+            color="toolbar-button"
+            textColor="toolbar-button-display"
+            class="text-no-wrap text-bold q-mr-sm"
+            :disabled="selectedTrack === null || projectFileError != undefined"
+            @click="handleImportTrack"
+          />
+        </QToolbar>
+      </QFooter>
+    </QLayout>
+  </QDialog>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from "vue";
+import { useDialogPluginComponent } from "quasar";
+import {
+  Project as UfProject,
+  EmptyProjectException,
+  IllegalFileException,
+  NotesOverlappingException,
+  supportedExtensions,
+  SupportedExtensions as UfSupportedExtensions,
+} from "@sevenc-nanashi/utaformatix-ts";
+import { useStore } from "@/store";
+import { createLogger } from "@/domain/frontend/log";
+import { ExhaustiveError } from "@/type/utility";
+import { IsEqual } from "@/type/utility";
+import { LatestProjectType } from "@/domain/project/schema";
+
+const { dialogRef, onDialogOK, onDialogCancel } = useDialogPluginComponent();
+
+const store = useStore();
+const log = createLogger("ImportExternalProjectDialog");
+
+// 受け入れる拡張子
+const acceptExtensions = computed(
+  () => supportedExtensions.map((ext) => `.${ext}`).join(",") + ",.vvproj",
+);
+
+type SupportedExtensions = UfSupportedExtensions | "vvproj";
+
+const projectNameToExtensions = [
+  ["VOICEVOX", ["vvproj"]],
+  ["MIDI (SMF)", ["mid"]],
+  ["MusicXML", ["xml", "musicxml"]],
+  ["CeVIO", ["ccs"]],
+  ["DeepVocal", ["dv"]],
+  ["OpenUtau", ["ustx"]],
+  ["Piapro Studio", ["ppsf"]],
+  ["Synthesizer V", ["s5p", "svp"]],
+  ["UtaFormatix", ["ufdata"]],
+  ["UTAU", ["ust"]],
+  ["VOCALOID", ["vpr", "vsq", "vsqx"]],
+] as const satisfies [string, SupportedExtensions[]][];
+
+// ちゃんと全部の拡張子があるかチェック
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const _: IsEqual<
+  (typeof projectNameToExtensions)[number][1][number],
+  SupportedExtensions
+> = true;
+
+// プロジェクトファイル
+const projectFile = ref<File | null>(null);
+// エラー
+const error = ref<
+  | "emptyProject"
+  | "overlapping"
+  | "parseFailed"
+  | "oldProject"
+  | "unknown"
+  | null
+>(null);
+
+// ファイルエラー
+const projectFileError = computed(() => {
```

Comment:
どっちがどっちかわからなそう
下はメッセージかな
```suggestion
// エラー
const error = ref<
  | "emptyProject"
  | "overlapping"
  | "parseFailed"
  | "oldProject"
  | "unknown"
  | null
>(null);

// エラーメッセージ
const projectFileErrorMessage = computed(() => {
```